backend/Comments/tests.py

```diff
@@ -1,3 +1,399 @@
+        data = {"message": "THIS IS A REPLY MESSAGE", "comment_reply_id": comment.id, "post_id": self.post.id  }
+        response = self.client.post(url, data, format="json")
+        # Use to debug: print("Create Comment with Reply Response:", response.data)  
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        
+    def test_comments_order_listed_by_likes(self):
+        com1 = Comment.objects.create(post=self.post, author=self.user, message="TEST Comment #1")
+        com2 = Comment.objects.create(post=self.post, author=self.user, message="TEST Comment #2")
+        # added like to Comment 1, 0 comments for Comment #2
+        com1.likes.add(self.user)
+
+        url = f"{reverse('comments:comment-list', kwargs={'post_id': self.post.id})}?order_by=likes"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data[0]['message'], "TEST Comment #1")
+            
+    def test_comment_replies_list(self):
+        comment = Comment.objects.create(post=self.post, author=self.user, message="Parent Comment")
+        Comment.objects.create(post=self.post, author=self.user, message="TEST Reply 1", comment_reply=comment)
+        url = reverse("comments:comment-list-create", kwargs={"comment_id": comment.id})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['message'], "TEST Reply 1")
+        
+    def test_like_reply(self):
+        reply = Comment.objects.create(author=self.user,post=self.post, message="TEST reply message", comment_reply=self.comment)
+        url = reverse("comments:like-comment", kwargs={"comment_id": reply.id})
+        response = self.client.patch(url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_dislike_reply(self):
+        reply = Comment.objects.create(author=self.user,post=self.post,message="Test reply message", comment_reply=self.comment)
+        url = reverse("comments:dislike-comment", kwargs={"comment_id": reply.id})
+        response = self.client.patch(url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+    
+    # Create a different user who is neither the author and should not have permission to delete  
+    def test_unauthorized_comment_delete_permission(self):
+        other_user = User.objects.create_user(username='other_user', password='other_user_passwird')
+        self.client.force_authenticate(user=other_user)
+        url = reverse("comments:comment-delete", kwargs={"comment_id": self.comment.id})
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        
+    def test_Moderator_deleting_comment(self):
```

Comment:
Small nitpick but you could probably de-capitalize Moderator and Hub in the titles of these two tests